src/ResourceManager/Compute/Commands.Compute/Strategies/ComputeRp/ImageEx.cs

```diff
@@ -65,8 +129,21 @@ public static int[] UpdatePorts(this OperatingSystemTypes osType, int[] ports)
             }
             else
             {
-                // get image
-                return Images
+                var compute = client.GetClient<ComputeManagementClient>();
```

Comment:
We shouldn't duplicate this code -it feels like this should be in the same if case as if the image contains a '/'

Reply:
I would suggest first checking if the name contains a ':', then a single case for '/' or no '/'.